tools/wpt/tests/test_update_expectations.py

```diff
@@ -0,0 +1,124 @@
+
+    def create_metadata(test_id, subtest_name, product, status="OK", subtest_status="PASS", channel="nightly"):
+        run_info = {
+            "os": "linux",
+            "processor": "x86_64",
+            "version": "Ubuntu 20.04",
+            "os_version": "20.04",
+            "bits": 64,
+            "linux_distro": "Ubuntu",
+            "product": product,
+            "debug": False,
+            "browser_version": "98.0.2",
+            "browser_channel": channel,
+            "verify": False,
+            "headless": True,
+        }
+
+        result = {
+            "test": test_id,
+            "subtests": [
+                {
+                    "name": subtest_name,
+                    "status": subtest_status,
+                    "message": None,
+                    "known_intermittent": []
+                }
+            ],
+            "status": status,
+            "message": None,
+            "duration": 555,
+            "known_intermittent": []
+        }
+
+        if status != "OK":
+            result["expected"] = "OK"
+
+        if subtest_status != "PASS":
+            result["subtests"][0]["expected"] = "PASS"
+
+        data = {
+            "time_start": 1648629686379,
+            "run_info": run_info,
+            "results": [result],
+            "time_end": 1648629698721
+        }
+
+        path = os.path.join(tmp_path, f"wptreport-{len(created_files)}.json")
+        with open(path, "w") as f:
+            json.dump(data, f)
+
+        created_files.append(path)
+        return run_info, path
+
+    yield create_metadata
+
+    for path in created_files:
+        os.unlink(path)
+
+
+def test_update(tmp_path, metadata_file):
+    # This has to be a real test so it's in the manifest
+    test_id = "/infrastructure/assumptions/cookie.html"
+    subtest_name = "cookies work in default browse settings"
+    test_path = os.path.join("infrastructure",
+                             "assumptions",
+                             "cookie.html")
+    run_info_firefox, path_firefox = metadata_file(test_id,
+                                                   subtest_name,
+                                                   "firefox",
+                                                   subtest_status="FAIL",
+                                                   channel="nightly")
+    run_info_chrome, path_chrome = metadata_file(test_id,
+                                                 subtest_name,
+                                                 "chrome",
+                                                 status="ERROR",
+                                                 subtest_status="NOTRUN",
+                                                 channel="dev")
+
+    metadata_path = str(os.path.join(tmp_path, "metadata"))
+    os.makedirs(metadata_path)
+    wptreport_paths = [path_firefox, path_chrome]
+
+    args = ["update-expectations",
+            "--manifest", os.path.join(repo_root, "MANIFEST.json"),
```

Comment:
Will this test work if MANIFEST.json doesn't exist yet? If not, it was probably created by some other integration test, but can we avoid that dependency?

Reply:
Yes, because the frontend updates the manifest which will create it if needed.

Reply:
Ah, I guess that happens in the `manifest_update` call.